File: test/language/expressions/python/ValueOfOperatorTest.py
import unittest

from testutils import getZserioApi

class ValueOfOperatorTest(unittest.TestCase):
    @classmethod
    def setUpClass(cls):
        cls.api = getZserioApi(__file__, "expressions.zs").valueof_operator

    def testGetValueOfWhiteColor(self):
        valueOfFunctions = self.api.ValueOfFunctions(self.api.Color.WHITE)
        whiteColorValue = 1
        self.assertEqual(whiteColorValue, valueOfFunctions.get_value_of_color())
        self.assertEqual(whiteColorValue, valueOfFunctions.get_value_of_white_color())

    def testGetValueOfBlackColor(self):
        valueOfFunctions = self.api.ValueOfFunctions(self.api.Color.BLACK)
        blackColorValue = 2
        self.assertEqual(blackColorValue, valueOfFunctions.get_value_of_color())
        self.assertEqual(blackColorValue, valueOfFunctions.get_value_of_black_color())

File: test/language/expressions/cpp/ValueOfOperatorTest.cpp
#include "math.h"

#include "gtest/gtest.h"

#include "expressions/valueof_operator/ValueOfFunctions.h"

namespace expressions
{
namespace valueof_operator
{

TEST(ValueOfOperatorTest, GetValueOfWhiteColor)
{
    ValueOfFunctions valueOfFunctions;
    valueOfFunctions.setColor(Color::WHITE);
    const uint8_t whiteColor = 1;
    ASSERT_EQ(whiteColor, valueOfFunctions.funcGetValueOfColor());
    ASSERT_EQ(whiteColor, valueOfFunctions.funcGetValueOfWhiteColor());
}

TEST(ValueOfOperatorTest, GetValueOfBlackColor)
{
    ValueOfFunctions valueOfFunctions;
    valueOfFunctions.setColor(Color::BLACK);
    const uint8_t blackColor = 2;
    ASSERT_EQ(blackColor, valueOfFunctions.funcGetValueOfColor());
    ASSERT_EQ(blackColor, valueOfFunctions.funcGetValueOfBlackColor());
}

} // namespace valueof_operator
} // namespace expressions
